addons/t9n/models/resource.py

```diff
@@ -1,11 +1,36 @@
-from odoo import fields, models
+from odoo import fields, models, api, _, Command
+from odoo.exceptions import ValidationError
+import polib
+import base64
+
+
+def decode_resource_file(resource_file):
+    try:
+        file_content = base64.b64decode(resource_file).decode("utf-8")
+        po_obj = polib.pofile(file_content)
+    except:
+        raise ValidationError(_("The file should be a .po file"))
+    return po_obj
+
+
+def create_messages(resource_file):
```

Comment:
I feel that the naming of this function is misleading. When a function is called "create_something", I expect it to have the side effect of actually creating something. This method does not create anything, it has no side effects